core/shared/src/main/scala/zio/Semaphore.scala

```diff
@@ -96,17 +96,31 @@ final class Semaphore private (private val state: Ref[State]) extends Serializab
   }
 
   /**
-   * Acquires a permit, executes the action and releases the permits right after.
+   * Acquires a permit, executes the action and releases the permit right after.
    */
   final def withPermit[R, E, A](task: ZIO[R, E, A]): ZIO[R, E, A] =
-    withPermits(1L)(task)
+    withPermits(1)(task)
+
+  /**
+   * Acquires a permit in a [[zio.ZManaged]], executes the action and releases the
+   * permit in the finalizer.
+   */
+  final def withPermitManaged[R, E, A](task: ZIO[R, E, A]): ZManaged[R, E, A] =
+    withPermitsManaged(1)(task)
 
   /**
    * Acquires `n` permits, executes the action and releases the permits right after.
    */
   final def withPermits[R, E, A](n: Long)(task: ZIO[R, E, A]): ZIO[R, E, A] =
     prepare(n).bracket(_.release)(_.awaitAcquire *> task)
 
+  /**
+   * Acquires `n` permits in a [[zio.ZManaged]], executes the action and releases the
+   * permits in the finalizer.
+   */
+  final def withPermitsManaged[R, E, A](n: Long)(task: ZIO[R, E, A]): ZManaged[R, E, A] =
```

Comment:
`task` is actually redundant here. This method should return a `ZManaged[R, E, Unit]` and the same semantics can be achieved by flatMapping the returned managed

Reply:
Awesome, will do.